assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+                    lectureList.add(lecture);
+                }
+            }
+            schedule.setLectureList(lectureList);
+        }
+----
+
+////
+[[useTheSolver]]
+[[theSolverInterface]]
+
+
+== The {PLANNER_SHORT} Solver
+A solver finds the best and optimal solution to your planning problem. A solver can only solve one planning problem instance at a time. Solvers are built with the `SolverFactory` method:
+
+[source,java,options="nowrap"]
+----
+public interface Solver<Solution_> {
+
+    Solution_ solve(Solution_ problem);
+
+    ...
+}
+----
+
+A solver should only be accessed from a single thread, except for the methods that are specifically documented in `javadoc` as being thread-safe. The `solve()` method hogs the current thread. This can cause HTTP timeouts for REST services and it requires extra code to solve multiple data sets in parallel. To avoid such issues, use a `SolverManager` instead.
+
+
+[[solvingAProblem]]
+=== Solving a problem
+
+You can use the solver to solve a planning problem.
+
+.Prerequisites
+* A `Solver` built from a solver configuration
+* An `@PlanningSolution` annotation that represents the planning problem instance
+
+.Procedure
+Provide the planning problem as argument to the `solve()` method. The solver will return the best solution found.
+
+The following example solves the NQueens problem:
+
+[source,java,options="nowrap"]
+----
+    NQueens problem = ...;
+    NQueens bestSolution = solver.solve(problem);
+----
+
+In this example, the `solve()` method will return an `NQueens` instance with every `Queen` assigned to a ``Row``.
+
+[NOTE]
+====
+The solution instance given to the `solve(Solution)` method does not need to be uninitialized. It can be partially or fully initialized, which is often the case in repeated planning.
+====
+
+.Best Solution for the Four Queens Puzzle in 8ms (Also an Optimal Solution)
+ifdef::OPTAPLANNER-COMM[]
+image::PlannerConfiguration/solvedNQueens04.png[align="left"]
+endif::OPTAPLANNER-COMM[]
+ifdef::OPTAPLANNER-ENT[]
+image::optimizer/solvedNQueens04.png[align="left"]
+endif::OPTAPLANNER-ENT[]
+
+
+The `solve(Solution)` method can take a long time depending on the problem size and the solver configuration. The `Solver` intelligently works through the search space of possible solutions and remembers the best solution it encounters during solving.
+Depending on a number of factors, including problem size, how much time the `Solver` has, the solver configuration, and so forth, that best solution might or might not be an optimal solution.
+
+[NOTE]
+====
+The solution instance given to the method `solve(solution)` is changed by the ``Solver``, but it might not be the best solution.
+
+The solution instance returned by the methods `solve(solution)` or `getBestSolution()` is most likely a planning clone of the instance given to the method ``solve(solution)``, which implies it is a different instance.
+====
```

Comment:
Is that intentional that the third note have been skipped

Reply:
It's not skipped, just moved up  to where I thought it related to the example.  Now line 272..

Reply:
LGTM